docs/spec/spec.md

```diff
@@ -0,0 +1,260 @@
+# Specification: Ballerina LDAP Library
+
+_Authors_: @Nuvindu \
+_Reviewers_: @NipunaRanasinghe @ayeshLK @DimuthuMadushan \
+_Created_: 2024/08/11 \
+_Updated_: 2024/08/11 \
+_Edition_: Swan Lake
+
+## Introduction
+
+This is the specification for the LDAP library of Ballerina language, which provides the capability to connect, authenticate, and interact with directory servers. It allows users to perform operations such as searching for entries, and modifying entries in an LDAP directory, providing support for directory-based operations.
+
+The LDAP library specification has evolved and may continue to evolve in the future. The released versions of the specification can be found under the relevant Github tag.
+
+If you have any feedback or suggestions about the library, start a discussion via a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues) or in the [Discord server](https://discord.gg/ballerinalang). Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in GitHub.
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [LDAP client](#2-ldap-client)
+    * 2.1 [Configurations](#21-configurations)
+    * 2.2 [Initialization](#22-initialization)
+    * 2.3 [DNs and RDNs](#23-dns-and-rdns)
+    * 2.4 [The `ldap:Entry` type](#24-the-ldapentry-type)
+    * 2.5 [The `ldap:LdapResponse` type](#25-the-ldapldapresponse-type)
+    * 2.6 [The `ldap:Error` type](#26-the-ldaperror-type)
+3. [Operation types](#3-operation-types)
```

Comment:
I propose the following structure for the `client` section for better readability and categorization. wdyt?
<img width="493" alt="Screenshot 2024-09-03 at 15 05 43" src="https://github.com/user-attachments/assets/57ac9ebd-09a9-45cd-b638-eca686054230">

Reply:
Fixed in 5de06c2